airflow/models/taskinstance.py

```diff
@@ -2730,6 +2730,14 @@ def tg2(inp):
         :return: Specific map index or map indexes to pull, or ``None`` if we
             want to "whole" return value (i.e. no mapped task groups involved).
         """
+        # This value should never be None since we already know the current task
+        # is in a mapped task group, and should have been expanded. The first check
+        # exists mainly to satisfy Mypy.
+        # But this value can be 0 when we expand an empty list, so th second check
+        # is necessary to avoid dividing by 0.
+        if ti_count is None or ti_count == 0:
```

Comment:
Can this be `if not ti_count`? (The comment can be modified slightly to explain why this is not simply `ti_count == 0`)

Reply:
Or if we want to keep the None and 0 checks separate, it may be better to split them into two separate `if` branches to make logic and comment easier to read.

Reply:
I don't see any problem in the first option, can you check my last commit?